File: CompetitiveProgramming/CCC/CCC '03 S1 - Snakes and Ladders/2020-11-20/SnakesAndLadders.py
import sys
data = sys.stdin.read().split('\n')

pos = 1
for i in range (len(data)):
	if int(data[i]) == 0:
		print("You Quit!")
		break
		
	if pos + int(data[i]) <= 100:
		pos += int(data[i])
	
	# snakes
	pos = 19 if pos == 54 else pos
	pos = 48 if pos == 90 else pos
	pos = 77 if pos == 99 else pos
	
	# ladders
	pos = 34 if pos == 9 else pos
	pos = 64 if pos == 40 else pos
	pos = 86 if pos == 67 else pos
		
	# output
	print("You are now on square", pos)
	
	# exit
	if pos == 100:
		print("You Win!")
		break

File: CompetitiveProgramming/CCC/CCC '03 S1 - Snakes and Ladders/2019-03-15/SnakesAndLadders.cpp
#include <iostream>
using namespace std;

int main()
{
	int square = 1;
	int input;
	
	while(1)
	{
		cin >> input;
		if(input == 0)
		{
			cout << "You Quit!";
			return 0;
		}
		square += input;
		
		switch(square)
		{
			case 9: square = 34; break;
			case 40: square = 64; break;
			case 67: square = 86; break;
			case 54: square = 19; break;
			case 90: square = 48; break;
			case 99: square = 77; break;
		}
		if(square > 100)
			square -= input;

		cout << "You are now on square " << square << endl;
		if(square == 100)
		{
			cout << "You Win!";
			return 0;
		}
		
	}
	
	return 0;
}